doc/en/user/source/community/oauth2/index.rst

```diff
@@ -0,0 +1,151 @@
+   that are known to both Google and your application. The set of values varies based on what type of application you are building. 
+   For example, a JavaScript application does not require a secret, but a web server application does.
+   
+    .. figure:: images/google1.jpg
+       :align: center
+
+#. Create a new ``OAuth client ID``
+
+   Click on the ``Client credentials`` context menu as shown in the figure below.
+   
+    .. figure:: images/google2.jpg
+       :align: center
+
+#. Configure a new ``Web application``
+
+   Add the ``Name`` and the ``Authorized redirect URIs`` like shown here below.
+   
+    .. note:: Always add two entries for each URI. One without the ending ``/`` and another one with it.
+   
+    .. figure:: images/google3.jpg
+       :align: center
+
+#. Take note of the ``Client ID`` and the ``Client Sectet``.
+
+   At the end of the procedure Google will show-up a small dialog box with the ``Client ID`` and the ``Client Sectet``.
+   Those info can be always accessed and updated from the `Google API Console <https://console.developers.google.com/>`_
+   
+    .. figure:: images/google4.jpg
+       :align: center
+
+#. Optionally customize the ``OAuth consent screen``.
+
+   At any time it is possible to update and customize the ``OAuth consent screen``. You can put here your logo, app name, ToS and so on.
+
+    .. figure:: images/google5.jpg
+       :align: center
+
+Configure the GeoServer OAuth2 filter
+-------------------------------------
+
+#. Start GeoServer and login to the web admin interface as the ``admin`` user.
+#. Click the ``Authentication`` link located under the ``Security`` section of
+   the navigation sidebar.
+
+    .. figure:: images/filter1.jpg
+       :align: center
+
+#. Scroll down to the ``Authentication Filters`` panel and click the ``Add new`` link.
+
+    .. figure:: images/filter2.jpg
+       :align: center
+
+#. Click the ``OAuth2`` link.
+
+    .. figure:: images/filter3.jpg
+       :align: center
+
+#. Fill in the fields of the settings form as follows:
+
+    .. figure:: images/filter4.jpg
+       :align: center
+   
+    .. note:: 
+	
+	   #. ``Scopes`` should be::
+	       
+		   https://www.googleapis.com/auth/userinfo.email,https://www.googleapis.com/auth/userinfo.profile
+		
+	   #. ``Client ID`` and ``Client Secret`` are the ones Google provided
+	   
+	   #. Choose a ``Role Service`` able to recognize user emails as IDs. By default a connected user will have ``ROLE_USER`` role
+	
+#. Update the filter chains by adding the new OAuth2 filter. 
+
+   .. figure:: images/filter5.jpg
+      :align: center
+
+#. Select the OAuth2 Filter for each filter chain you want to protect with OAuth2.
+
+   .. figure:: images/filter6.jpg
+      :align: center
+
+   Be sure to select and order correctly the OAuth2 Filter.
+
+#. Save.
+
+Test a Google OAuth2 login
+--------------------------
+
+#. Navigate to the GeoServer home page and log out of the admin account. 
```

Comment:
Something went wrong here, and I cannot login anymore, google makes me login and then brings me to the account settings page instead of the geoserver page. My guess is that it's due to the redirect url. Some instructions to un-brick GeoServer would be useful.
In my case I went to modify GeoServer config at $GEOSERVER_DATA_DIR/release/security/filter/oauth2-google and set the redirect URI to one of the authorized ones, restarted and then I get automatically logged in.
However, GeoServer is still "bricked", as I can only login with my google account now, no way to get to the home page as anonymous, and no way to get back as admin using the local authentication. I believe this needs to be solved, or instructions on how to make sure admin control is not lost.